LanguageExt.Core/DataTypes/Seq/Seq.cs

```diff
@@ -808,20 +808,14 @@ public override int GetHashCode()
         /// <summary>
         /// Equality test
         /// </summary>
-        public bool Equals(Seq<A> rhs)
-        {
-            var x = this;
-            var y = rhs;
-
-            if (x.seq == null && y.seq == null &&
-                (x.Count != y.Count ||
-                 x.GetHashCode() != y.GetHashCode()))
-            {
-                return false;
-            }
-
-            return Enumerable.SequenceEqual(this, rhs);
-        }
+        public bool Equals(Seq<A> rhs) =>
+            !ReferenceEquals(rhs, null) &&
+            this.seq == null && rhs.seq == null &&
+            this.Count == rhs.Count &&
+            this.GetHashCode() == rhs.GetHashCode() &&
+            Enumerable.SequenceEqual(this, rhs)
+                 ? true
```

Comment:
Why is this ternary here? Are we just using this to make sure that exceptions or side-effect that might be happen deep in `Enumerable.SequenceEqual` get a chance?